### DISCUSSION TITLE
Docs: "Transform" - Column Model

### DISCUSSION THREAD

**Discussion by andy-bond** [shlomiassaf/ngrid#46] [Q&A] [answered]
#### Please describe the feature you would like to request.

I have been using the `transform` feature on the column model and have found it very useful, but I can't actually find it anywhere in the documentation. I just happened to stumble upon it when looking at #4 .

#### What is the use-case or motivation for this proposal?

I think it is a very useful feature and I think people should know it exists.

#### Is there anything else we should know?

I don't fully understand what the first argument of the `transform` function is used for, I tend to not need it:
```
{
    prop: 'amendments',
    transform: (amendments, bill: Bill) => this.getBillAmendments(bill),
    label: 'AMENDMENTS',
    sort: this.sort,
    width: '250px'
},
```

**Comment by shlomiassaf** [comment]
Hi, thanks for the heads up.

The `transform` property is documented in the code, see [source code](https://github.com/shlomiassaf/ngrid/blob/master/libs/ngrid/src/lib/table/columns/column.ts#L131)

If you're using VSCode or Webstore, you can view it by going to (or peeking) the definitions.

```ts
  /**
   * Optional transformer that control the value output from the combination of a column and a row.
   * The value returned from this transformer will be returned from `PblColumn.getValue`
   */
  transform?: (value: any, row?: any, col?: PblColumn) => any;
```

The first param, **value** is the value of the cell your are transforming.

The 2nd param, **row** is the entire row.

The 3rd param, **col** is the `PblColumn` instance.

So if your in column "name"

`value === row.name === col.getValue(row)` // DON'T DO THIS :)

Now, you wonder, why providing value if you can do `row.name` or `col.getValue(row)`?

First, `row.name` is the raw value, not transformed and not processed so it's not good.

2nd, `col.getValue(row)` will call `transform` so you can't call it from inside of transform or you will get into a loop and error.

The value is just the value of the cell before it was transformed... that is all.

I will add this to the docs sometime soon.

**Comment by shlomiassaf** [comment, marked as answer]
Document in [here](https://shlomiassaf.github.io/ngrid/features/column/column-transform)
